components/dashboard/src/start/WorkspaceLogs.tsx

```diff
@@ -4,8 +4,81 @@
  * See License-AGPL.txt in the project root for license information.
  */
 
-export default function WorkspaceLogs() {
-  return <pre className="m-6 p-4 h-72 w-11/12 lg:w-3/5 flex-shrink-0 rounded-xl bg-gray-100">
-    ... logs ...
-  </pre>;
+import EventEmitter from 'events';
+import React from 'react';
+import { Terminal, ITerminalOptions, ITheme } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit'
+import 'xterm/css/xterm.css';
+import { DisposableCollection } from '@gitpod/gitpod-protocol';
+
+export interface WorkspaceLogsProps {
+  logsEmitter: EventEmitter;
+}
+
+export interface WorkspaceLogsState {
+}
+
+export default class WorkspaceLogs extends React.Component<WorkspaceLogsProps, WorkspaceLogsState> {
+  protected xTermParentRef: React.RefObject<HTMLDivElement>;
+  protected terminal: Terminal | undefined;
+  protected fitAddon: FitAddon | undefined;
+  
+  constructor(props: WorkspaceLogsProps) {
+    super(props);
+    this.xTermParentRef = React.createRef();
+  }
+
+  private readonly toDispose = new DisposableCollection();
+  componentDidMount() {
+    const element = this.xTermParentRef.current;
+    if (element === null) {
+      return;
+    }
+    const theme: ITheme = {
+      // background: '#F5F5F4',
+    };
+    const options: ITerminalOptions = {
+      cursorBlink: false,
+      disableStdin: true,
+      fontSize: 14,
+      // fontWeight: 'normal',
+      theme,
+      scrollback: 9999999,
+    };
+    this.terminal = new Terminal(options);
+    this.fitAddon = new FitAddon();
+    this.terminal.loadAddon(this.fitAddon);
+    this.terminal.open(element);
+    this.props.logsEmitter.on('logs', logs => {
+      if (this.fitAddon && this.terminal && logs) {
+        this.terminal.write(logs);
+      }
+    });
+    this.toDispose.push(this.terminal);
+    this.fitAddon.fit();
+
+    // Fit terminal on window resize (debounced)
+    let timeout: NodeJS.Timeout | undefined;
+    const onWindowResize = () => {
+      clearTimeout(timeout!);
+      timeout = setTimeout(() => this.fitAddon!.fit(), 20);
+    };
+    window.addEventListener('resize', onWindowResize);
+    this.toDispose.push({
+      dispose: () => {
+        clearTimeout(timeout!);
+        window.removeEventListener('resize', onWindowResize);
+      }
+    });
+  }
+
+  componentWillUnmount() {
+    this.toDispose.dispose();
+  }
+
+  render() {
+    return <div className="mt-6 h-72 w-11/12 lg:w-3/5 rounded-xl bg-black p-6">
```

Comment:
**question**: Is `w-11/12` needed here?

Reply:
I doesn't have much an effect on iPad width, but improves mobile width:

<table><tr><th>With <code>w-11/12</code></th><th>Without <code>w-11/12</code></th></tr><tr><td>
<img width="384" alt="Screenshot 2021-03-26 at 08 57 00" src="https://user-images.githubusercontent.com/599268/112600416-772da380-8e11-11eb-9708-399f1d035d07.png">
</td><td>
<img width="385" alt="Screenshot 2021-03-26 at 08 56 33" src="https://user-images.githubusercontent.com/599268/112600412-75fc7680-8e11-11eb-9320-918d52475831.png">
</td></tr></table>

Also, while I agree mobile is out-of-scope, I don't see the value of breaking it on purpose by removing `w-11/12`. 😇

Reply:
DEAL. Let's leave this!